specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-04-01-preview/providers.json

```diff
@@ -152,6 +159,81 @@
           "description": "Blob duration of specification."
         }
       }
+    },
+    "MetricSpecification": {
+      "description": "Metric specification of operation.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of metric specification."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display name of Metric specification."
+        },
+        "displayDescription": {
+          "type": "string",
+          "description": "Display description of Metric specification."
+        },
+        "unit": {
+          "type": "string",
```

Comment:
just curious, why use string instead of enum?

Reply:
The whole specification metric is read only for customer. We expose these metrics via public metrics REST API. The values of these properties are possible values. They are not the only values as far as I know.